### PR TITLE
FLE expand convergence in unit test problems.

### PR DESCRIPTION
**Pull request by garrettwrong**
Looking into another flaky test (#862).

### CODE REVIEW

**Comment by garrettwrong** [comment]
With this change I have not been able to reproduce locally after several hundred runs.  Will see if envountered in any of the CI jobs.

**Comment by garrettwrong** [comment]
Was one CI fail, but it was for micrograph sim, not FLE.

File: tests/test_FLEbasis2D.py [hunk 1]
```diff
@@ -55,7 +55,9 @@ def create_images(L, n):
         np.load(os.path.join(DATA_DIR, "clean70SRibosome_vol.npy")).astype(np.float64)
     )
     v = v.downsample(L)
-    sim = Simulation(L=L, n=n, vols=v, dtype=v.dtype, seed=1103)
+    sim = Simulation(
+        L=L, n=n, vols=v, dtype=v.dtype, offsets=0, amplitudes=1, seed=1103
+    )
```
Comment (j-c-c) [inline tests/test_FLEbasis2D.py:60]: Pesky offsets!

**Review by j-c-c — DISMISSED** [review]
Looks good. Will be nice to get some more info in the event of a future failure.

File: src/aspire/basis/basis.py [hunk 2]
```diff
@@ -538,6 +538,9 @@ def expand(self, x):
 
         :param x: An array whose last two or three dimensions are to be expanded
             the desired basis. These dimensions must equal `self.sz`.
+        :param tol: Tolerances for convergence, `norm(residual) <= max(tol*norm(b), atol)`.
```
Comment (janden) [inline src/aspire/basis/basis.py]: Maybe reword this to `Relative tolerance for convergence` and next one to `Absolute tolerance for convergence`?

**Review by janden — CHANGES_REQUESTED** [review]
I'm confused. We introduce the `tol` and `atol` arguments to `expand`, but this does not appear to be used anywhere (default arguments reproduce old behavior). Is the fix just setting offsets to zero and amplitudes to one?

File: src/aspire/basis/basis.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/basis/basis.py]: Sure, sounds good.

**Comment by garrettwrong** [comment]
@janden , yes, fixing the Simulation params seemed to fix the sporadic non convergence.  I can't say that 100% fixes all occurrences, but it reduced them to the level I have been unable to reproduce on our dev machine so far. I'd like to take this patch to see... the degrading CI robustness has been on my nerves for a while now.

Regarding the other changes.  First, I really want to keep the emission of the return code in the error message. 
That is important.  

I can remove the addition of the `tol` and `atol` args if you prefer.  I found them useful when debugging.  Also I'm not entirely sure how carefully those cg constraints for `expand` were actually chosen, so there is a good chance they need to be tweaked in practice.  I think this has happened with at least one post doc while I worked here.  Let me know... made the docstring updates, but I'd be happy to remove as well.

**Comment by janden** [comment]
> Regarding the other changes. First, I really want to keep the emission of the return code in the error message. That is important.
> 
> I can remove the addition of the `tol` and `atol` args if you prefer. I found them useful when debugging. Also I'm not entirely sure how carefully those cg constraints for `expand` were actually chosen, so there is a good chance they need to be tweaked in practice. I think this has happened with at least one post doc while I worked here. Let me know... made the docstring updates, but I'd be happy to remove as well.

No I just wanted to make sure I understood what was going on. Looks great.

**Review by janden — APPROVED** [review]
